### PR TITLE
Add chromatic visual tests

### PR DESCRIPTION
**Pull request by tmeasday**
Issue: We didn't have visual regression tests for our stories

Note: this is built on https://github.com/storybooks/storybook/pull/2504

## What I did

Added [chromatic](https://blog.hichroma.com/introducing-chromatic-ui-testing-for-react-c5cc01a79aaa) to our CI

## How to test

Make a UI change, check out the build at https://chromaticqa.com

<strike>**NOTE**: this is not currently accessible to members of the `storybooks` org, just sorting out org permissions.</strike>

Anyone in the `storybooks` org on GH should be able to access the project here: https://www.chromaticqa.com/builds?appId=5a375b97f4b14f0020b0cda3

### CODE REVIEW

**Comment by tmeasday** [comment]
@ndelangen do you have the permission to grant access from the `storybooks` GH org?

**Comment by Hypnosphi** [comment]
Note that there's an ongoing work on adding visual testing functionality to storyshots: #2413

**Comment by tmeasday** [comment]
@Hypnosphi 

> Note that there's an ongoing work on adding visual testing functionality to storyshots

Yes! I think this is great! I think however for a larger project like this one Chromatic can offer a little more on top of that, namely parallelized / cloud screenshots and an review flow UI. Interested to hear your thoughts.

File: package.json
Comment (igor-dv) [inline package.json]: If it's not a public key, probably better hide it with the CI parameter

Comment (tmeasday) [reply package.json]: Yeah, see the note on the commit ;)

Comment (igor-dv) [reply package.json]: Hah, 😄

**Comment by Hypnosphi** [comment]
Oh, that's actually pretty cool. I think there's a value in adding this for all the `cra-kitchen-sink` stories

**Comment by tmeasday** [comment]
> I think there's a value in adding this for all the cra-kitchen-sink stories

Yes, I'll get this going once we sort out #2504

**Comment by ndelangen** [comment]
Please rebase, so i can better judge this PR, looks all right.

**Comment by tmeasday** [comment]
@ndelangen done.

File: .circleci/config.yml
Comment (Hypnosphi) [inline .circleci/config.yml]: Please add a root-level shortcut script for that

Comment (Hypnosphi) [reply .circleci/config.yml]: Maybe it should be even made an option in `scripts/test.js`

Comment (ndelangen) [reply .circleci/config.yml]: Is this something we want to be able to run locally?

Comment (Hypnosphi) [reply .circleci/config.yml]: I believe so, at least to update the refs

Comment (ndelangen) [reply .circleci/config.yml]: I was under the impression chromatic provided a review system to decide if a change is OK or NOT_OK ?

Comment (ndelangen) [reply .circleci/config.yml]: If the tests are run by each dev, how does chromatic overcome the rendering differences per system?

Comment (Hypnosphi) [reply .circleci/config.yml]: AFAIU they actually run in cloud even when launched locally

Comment (Hypnosphi) [reply .circleci/config.yml]: > chromatic provided a review system to decide if a change is OK or NOT_OK

That's cool that they do that! And I feel convenient doing that even before submitting a PR

Comment (tmeasday) [reply .circleci/config.yml]: Yeah, you can run it locally and it does take the snapshots in the cloud still.

The only hiccup with this is the `CHROMATIC_APP_CODE` environment variable -- it will need to be set for the test to run locally. Any thoughts on how to make that easier for devs (you can find it in the settings page at https://www.chromaticqa.com/settings?appId=5a375b97f4b14f0020b0cda3)

**Review by ndelangen — APPROVED** [review]

File: .circleci/config.yml
Comment (tmeasday) [reply .circleci/config.yml]: @Hypnosphi -- not sure how to best add it to the `scripts/test.js` command; it seems the way that is written a command either needs to be a jest project or a `npm test` command for some subdirectory. I'm not sure we want to make it `npm test` for `examples/official-storybook` though.

Should I change the script to support a full custom command? Perhaps in a separate PR?

**Comment by tmeasday** [comment]
This is ready to merge when the 3.3 release process is complete.

**Comment by Hypnosphi** [comment]
How does it affect release process?

**Comment by tmeasday** [comment]
It shouldn't, but @shilman asked me to hold off merging into `master` until the release is out...

**Comment by ndelangen** [comment]
Merge when ready, if you ask me

**Pull request merged**
